src/module.c

```diff
@@ -9149,7 +9402,8 @@ int RM_ACLAddLogEntry(RedisModuleCtx *ctx, RedisModuleUser *user, RedisModuleStr
         default: return REDISMODULE_ERR;
     }
 
-    addACLLogEntry(ctx->client, acl_reason, ACL_LOG_CTX_MODULE, -1, user->user->name, sdsdup(object->ptr));
+    sds obj = object ? sdsdup(object->ptr) : NULL;
```

Comment:
i guess we need to document this change / feature.
first, so that people know they can pass NULL, and secondly, so people who use it will understand it's incompatible with old versions.

Reply:
I updated the documentation here: https://github.com/redis/redis/pull/11659/commits/e9b0de469ca57b0c10b381c9734d5d6a6c8be0a2

Reply:
i think we need to document the version from which we started doing that, so module authors know that if they rely on that, their module isn't compatible with older versions.
@yossigo @madolson WDYT?

Reply:
I guess I still think I would rather the module be fully explicit here and pass in an object. It seems like that might be more useful for then authentication failure on HELLO, when it really failed on some module callback.

Reply:
I reverted changes to this API here - https://github.com/redis/redis/pull/11659/commits/35e0e6714d93b2dbcc876eb7289a228d15bb133a